### PR TITLE
added trigger parameters to the trigger plugin

### PR DESCRIPTION
**Pull request by geekodour**
an attempt at adding trigger ~time~ parameters to the trigger plugin

I am not sure if this is planned to be added but this can be way to implement this if required.

Signed-off-by: Hrishikesh Barman <hrishikeshbman@gmail.com>

### CODE REVIEW

File: prow/plugins/trigger/generic-comment.go [hunk 1]
```diff
@@ -114,6 +115,17 @@ func handleGenericComment(c Client, trigger plugins.Trigger, gc github.GenericCo
 	if err != nil {
 		return err
 	}
+
+	for _, ps := range toTest {
+		groups := ps.GetGroups(gc.Body)
+		if groups != nil {
+			for i, group := range groups[1:] {
+				env := v1.EnvVar{Name: fmt.Sprintf("ARG%d", i), Value: group}
```
Comment (fejta) [inline prow/plugins/trigger/generic-comment.go]: I don't understand how ARG0=group results in the trigger time?

Comment (geekodour) [reply prow/plugins/trigger/generic-comment.go]: eg regex for trigger field in prowjob:
```
trigger: "`(?mi)^/testcommand\s+([0-9])\s*$" 
```
so `groups[0]` will have the first matching group? that we put into `ARG0` env var and so on.

Comment (geekodour) [reply prow/plugins/trigger/generic-comment.go]: made this PR as part of this discussion, so we might mean different things by `trigger time parameters` i just meant having to pass the comment parameters to the job when it triggers.

https://kubernetes.slack.com/archives/CDECRSC5U/p1560940499239400

File: prow/config/jobs.go [hunk 2]
```diff
@@ -368,6 +368,11 @@ func (ps Presubmit) TriggerMatches(body string) bool {
 	return ps.Trigger != "" && ps.re.MatchString(body)
 }
 
+// GetGroups returns a slice of matched groups.
+func (ps Presubmit) GetGroups(body string) []string {
+	return ps.re.FindStringSubmatch(body)
```
Comment (fejta) [inline prow/config/jobs.go]: It looks like re may be nil

File: prow/plugins/trigger/generic-comment.go (hunk 1 above)
Comment (fejta) [reply prow/plugins/trigger/generic-comment.go]: "Time" makes me think this is about timestamps but it seems like you want to be able to send arbitrary parameters to triggered jobs.

Can you update the PR description to remove the word time? As Steve mentioned in this thread let's see if we can get consensus on the design of this feature before we do anything with it.

**Comment by stevekuznetsov** [comment]
I think barring a larger effort to fully consider the implications of such a change with a design document and review, this pull request will not move forward as it is making a large, fundamental change to how Prow runs jobs. If you do write a design doc, please send it out to the mailing lists!

/close

**Pull request closed without merging**